### ISSUE TITLE
Error: Actions must be plain objects when downloading sketches as zip

### ISSUE THREAD

**Issue by adadgarg** [processing/p5.js-web-editor#2311]
### What is your operating system?

Windows

### Web browser and version

Chrome version :114.0.5735.198

### Actual Behavior

![1](https://github.com/processing/p5.js-web-editor/assets/104292766/757387ce-0be9-4b1f-91fe-e84244c2c5d5)
The error occurs while downloading the sketch zip file. Although the zip file gets downloaded, but it shows the Error: "Actions must be plain objects. Use custom middleware for async actions."
I think the error occurs due to the attempt to dispatch an asynchronous action directly to the Redux store without using any middleware.

### Expected Behavior

Error should not occur.

### Steps:
1. Go to File tab.
2. Examples then download any sketch by using the down arrow button.

**Comment by lindapaiste** [comment]
Thanks for reporting this!  We do have thunk middleware but in this case we are dispatching an `action` variable is `undefined`.  That's because the `exportProjectAsZip` action creator function doesn't return anything.

https://github.com/processing/p5.js-web-editor/blob/3493b3b8099d42cac7220f0181ba4fbafebddbf0/client/modules/IDE/actions/project.js#L261-L264

 It doesn't involve Redux at all and does not need to be dispatched.  
 
 Lot of options here:
1. Call the function directly rather than dispatching it.
2. Keep the component code the same (keep the dispatch) and write the action creator as a thunk.
3. Return an action object from the action creator, even if it's not handled by a reducer (I don't love this).
4. Get rid of the `exportProjectAsZip` function and link directly to the download URL with the [`download` attribute](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/download).

Edit: 
I played with option 4 and I do think the downloading is smoother without opening a secondary window but I need to tweak some styles so that the `button` styles also apply to `a`.
![image](https://github.com/processing/p5.js-web-editor/assets/28965286/ccf946aa-0a43-48d4-9b21-f0c736115013)

Also realized that I implemented option 1 when reworking the `Nav`.
https://github.com/processing/p5.js-web-editor/blob/3493b3b8099d42cac7220f0181ba4fbafebddbf0/client/components/Nav.jsx#L70-L73
